website/docs/r/argo_tunnel.html.markdown

```diff
@@ -0,0 +1,44 @@
+---
+layout: "cloudflare"
+page_title: "Cloudflare: cloudflare_argo_tunnel"
+sidebar_current: "docs-cloudflare-resource-argo-tunnel"
+description: |-
+  Provides the ability to manage Cloudflare Argo Tunnels.
+---
+
+# cloudflare_argo_tunnel
+
+Argo Tunnel exposes applications running on your local web server on any network with an internet connection without manually adding DNS records or configuring a firewall or router.
+
+## Example Usage
+
+```hcl
+resource "cloudflare_argo_tunnel" "example" {
+  account_id = "d41d8cd98f00b204e9800998ecf8427e"
+  name       = "my-tunnel"
+  secret     = "AQIDBAUGBwgBAgMEBQYHCAECAwQFBgcIAQIDBAUGBwg="
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `account_id` - (Required) The Cloudflare account ID that you wish to manage the Argo Tunnel on.
+* `name` - (Required) A user-friendly name chosen when the tunnel is created. Cannot be empty.
+* `secret` - (Required) 32 or more bytes, encoded as a base64 string. The Create Argo Tunnel endpoint sets this as the tunnel's password. Anyone wishing to run the tunnel needs this password.
```

Comment:
Just a heads up..I'm not sure how the cloudflare-go library handles this, but the HTTP API does not have the requirement that the secret be a base64 string (it might not even have the requirement that it be 32 bytes or more, but all my scripts use 32 bytes or more so I can't say off the top of my head).

When the secret is auto generated by the cloudflared cli tool, it generates a base64 encoded string, but I think that is just their quick way of generating a random string.

I've used long strings generated with password generators with no problem.

Reply:
The requirement is [documented](https://api.cloudflare.com/#argo-tunnel-create-argo-tunnel) (also where I stole this line from) and is enforced by the API schema.

In terms of `cloudflare-go` handles this, it only passes on the string it is sent. It does not check for base64 encoding or the length. Instead, it will error out if the API errors out. Same principle here where we don't actually enforce it and instead offload that to the API itself.

Reply:
Huh, interesting. Thanks for the link @jacobbednarz. Looks like a mistake on our side, however I can report that the CF API (at least before christmas, when I last poked it) doesn't error out if you pass an invalid base64 string. Will update our code though to comply with the docs in the event it starts being enforced.